containers/src/Kokkos_DualView.hpp

```diff
@@ -1015,7 +1003,7 @@ class DualView : public ViewTraits<DataType, Properties...> {
         resync_host(properties);
 
         /* Mark Device copy as modified */
-        ++modified_flags(1);
+        modified_flags(1) = true;
```

Comment:
I think this is not sufficient. Don't you also `need modified_flags(0)=false`?

Reply:
I made sure that all modifications to `modified_flags` (apart from allocations) go through `modify` or `clear_sync_state` now. So we check that only one of them is set in `Debug` mode.

We only execute `resize_on_device` after calling `sync<typename t_dev::memory_space>(exec_space);` or if `modified_flags(0)` is not set. Similarly `resize_on_host` is only executed after calling `sync<typename t_host::memory_space>(exec_space);` or if `modified_flags(0)` is set.